File: packages/dataframe/rust/stats/distributions/normal.test.R
#!/usr/bin/env Rscript

# Normal distribution test script
# Usage: Rscript normal.test.R <function> <x_or_p> <mean> <sd> [lower_tail] [log_p]

source("test-helpers.R")

# Define normal distribution functions
normal_density <- function(x, mean, sd, log = FALSE) {
  dnorm(x, mean, sd, log = log)
}

normal_cumulative <- function(x, mean, sd, lower.tail = TRUE, log.p = FALSE) {
  pnorm(x, mean, sd, lower.tail = lower.tail, log.p = log.p)
}

normal_quantile <- function(p, mean, sd, lower.tail = TRUE, log.p = FALSE) {
  qnorm(p, mean, sd, lower.tail = lower.tail, log.p = log.p)
}

normal_random <- function(n, mean, sd) {
  rnorm(n, mean, sd)
}

# Run the test
run_distribution_test(normal_density, normal_cumulative, normal_quantile, normal_random)

File: packages/dataframe/rust/stats/distributions/normal.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust normal functions...\n");

  const testCases = [
    { func: "dnorm", distribution: "normal", args: [0, 0, 1, 0] },
    { func: "pnorm", distribution: "normal", args: [0, 0, 1, 1, 0] },
    { func: "qnorm", distribution: "normal", args: [0.5, 0, 1, 1, 0] },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test rnorm separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const rnormResult = await callR("rnorm", "normal", 1, 0, 1);
    console.log(`rnorm(0, 1) = ${rnormResult}`);
  } catch (error) {
    console.log(`rnorm: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
